azimuth/modules/dataset_analysis/similarity_analysis.py

```diff
@@ -73,30 +73,27 @@ class NeighborsTaggingModule(DatasetResultModule[SimilarityConfig]):
     """Compute neighbors for each utterance in a dataset split."""
 
     def compute_on_dataset_split(self) -> List[TaggingResponse]:  # type: ignore
-        if DatasetSplitName.train in self.available_dataset_splits:
-            train_dm: Optional[DatasetSplitManager] = self.get_dataset_split_manager(
-                DatasetSplitName.train
-            )
-            # NOTE: Doing this ensure that the FAISS index exists somewhere.
-            # If FAISSModule fails this module will fail instead of blocking.
-            train_features: Optional[List[Array]] = self._get_features(DatasetSplitName.train)
-        else:
-            train_dm = None
-            train_features = None
-
-        eval_dm = self.get_dataset_split_manager(DatasetSplitName.eval)
-        eval_features = self._get_features(DatasetSplitName.eval)
-        ds_features = assert_not_none(
-            eval_features if self.dataset_split_name == DatasetSplitName.eval else train_features
+        # Get the features from FAISS for all available splits to make sure they are all computed
+        # and that we can compute neighbors for all splits.
+        eval_features = (
+            self._get_features_from_faiss(DatasetSplitName.eval)
+            if DatasetSplitName.eval in self.available_dataset_splits
+            else None
+        )
+        train_features = (
+            self._get_features_from_faiss(DatasetSplitName.train)
+            if DatasetSplitName.train in self.available_dataset_splits
+            else None
```

Comment:
Would that work?
```suggestion
        features_dict = {
            split: self._get_features_from_faiss(split) for split in self.available_dataset_splits
        }
```
and
```diff
-        features = assert_not_none(
-            eval_features if self.dataset_split_name == DatasetSplitName.eval else train_features
-        )
+        features = features_dict[self.dataset_split_name]
```

Reply:
Woah, next level! I didn't think of a `Dict`. The tests pass! Awesome thanks!!

Reply:
Awesome! 👍